src/FindStmtCost.cpp

```diff
@@ -0,0 +1,1025 @@
+        } else {
+            cost = it->second.computation_cost_exclusive;
+        }
+    }
+
+    internal_assert(cost >= 0) << "\n"
+                               << "FindStmtCost::get_computation_cost: " << print_node(node)
+                               << "computation_cost_exclusive not set (cost is: " << cost << ")"
+                               << "\n\n";
+
+    return cost;
+}
+int FindStmtCost::get_data_movement_cost(const IRNode *node, bool inclusive) const {
+    internal_assert(node != nullptr) << "\n"
+                                     << "FindStmtCost::get_data_movement_cost: node is nullptr"
+                                     << "\n\n";
+
+    auto it = stmt_cost.find(node);
+    IRNodeType type = node->node_type;
+    int cost = -1;
+
+    if (it == stmt_cost.end()) {
+        // sometimes, these constant values are created on the whim in
+        // StmtToViz.cpp - set cost_node to be fresh StmtCost to avoid crashing
+        if (type == IRNodeType::IntImm || type == IRNodeType::UIntImm ||
+            type == IRNodeType::FloatImm || type == IRNodeType::StringImm) {
+            cost = NORMAL_NODE_DMC;
+        }
+
+        // this happens when visualizing cost of else_case in StmtToViz.cpp
+        else if (type == IRNodeType::Variable) {
+            const Variable *var = (const Variable *)node;
+            if (var->name == StmtToViz_canIgnoreVariableName_string) {
+                cost = NORMAL_NODE_DMC;
+            }
+        } else {
+            internal_assert(false) << "\n"
+                                   << "FindStmtCost::get_data_movement_cost: " << print_node(node)
+                                   << "node not found in stmt_cost map"
+                                   << "\n\n";
+            return 0;
+        }
+    } else {
+        if (inclusive) {
+            cost = it->second.data_movement_cost_inclusive;
+        } else {
+            cost = it->second.data_movement_cost_exclusive;
+        }
+    }
+
+    internal_assert(cost >= 0) << "\n"
+                               << "FindStmtCost::get_data_movement_cost: " << print_node(node)
+                               << "data_movement_cost_exclusive not set (cost is: " << cost << ")"
+                               << "\n\n";
+
+    return cost;
+}
+
+int FindStmtCost::get_if_node_cost(const IRNode *op, bool inclusive, bool is_computation) const {
+    internal_assert(op->node_type == IRNodeType::IfThenElse) << "\n"
+                                                             << "FindStmtCost::get_if_node_cost: " << print_node(op)
+                                                             << "node is not IfThenElse"
+                                                             << "\n\n";
+
+    int cost;
+    const IfThenElse *if_then_else = dynamic_cast<const IfThenElse *>(op);
+
+    if (is_computation) {
+        if (inclusive) {
+            int computation_cost = get_computation_cost(if_then_else->condition.get(), inclusive) +
```

Comment:
Unclear why inclusive cost ignores ignores the else-case. Discussing the cost-model is perhaps tangential to this PR but we should add a comment explaining the intention here.